enginetest/evaluation.go

```diff
@@ -538,6 +580,16 @@ func checkResults(
 				}
 				widenedExpected[i][j] = actual // ensure it passes equality check later
 			}
+
+			if isServerEngine {
+				if b, isBool := widenedExpected[i][j].(bool); isBool {
+					if b {
+						widenedExpected[i][j] = int64(1)
+					} else {
+						widenedExpected[i][j] = int64(0)
+					}
+				}
+			}
```

Comment:
This would be another good spot to document a cleanup TODO. We want to be pragmatic about getting the tests running, so doing some conversions like this totally makes sense, but would be good to leave a breadcrumb trail with TODOs that explain why we did these shortcuts and what the next step is to unravel them.